src/Compilers/CSharp/Test/Emit2/Semantics/LockTests.cs

```diff
@@ -1899,17 +2014,443 @@ public void AsyncMethod_WithAwait()
 
             class C
             {
-                async void M()
+                static async Task Main()
                 {
                     await Task.Yield();
-                    lock (new Lock()) { }
+                    lock (new Lock()) { System.Console.Write("L"); }
                 }
```

Comment:
>} [](http://example.com/codeflow?start=16&length=1)

Consider adding `await Task.Yield();` after the `lock (new Lock()) { ... }` as well.